kedro/io/data_catalog.py

```diff
@@ -311,6 +387,53 @@ def _get_dataset(
 
         return data_set
 
+    def __contains__(self, data_set_name):
```

Comment:
IMO `__contains__` should really only check if a dataset is in the catalog or not, but not add it. I would move that responsibility to the `_get_dataset` method instead. Mainly because `__contains__` is a magic function and is meant for implementing the membership operation, whereas now it does more than just that.

Reply:
Moved it to `_get_dataset`